projects/plugins/boost/app/assets/src/js/features/critical-css/lib/critical-css-errors.ts

```diff
@@ -159,32 +171,28 @@ export function groupRecommendationsByStatus(
 		const providerErrors = provider.errors || [];
 		// Group errors by type first
 		const errorsByType = providerErrors.reduce( ( acc, error ) => {
-			if ( ! acc[ error.type ] ) {
-				acc[ error.type ] = [];
+			const errorTypeKey = getErrorTypeKey( error );
+			if ( ! acc[ errorTypeKey ] ) {
+				acc[ errorTypeKey ] = [];
 			}
-			acc[ error.type ].push( error );
+			acc[ errorTypeKey ].push( error );
 			return acc;
 		}, {} as ErrorsByType );
 
-		const errorTypeGroups = Object.entries( errorsByType ) as [
-			Critical_CSS_Error_Type,
-			CriticalCssErrorDetails[],
-		][];
-
 		// For each error type group, check if it's dismissed
-		errorTypeGroups.forEach( ( [ errorType, errors ] ) => {
+		Object.entries( errorsByType ).forEach( ( [ errorType, errors ] ) => {
 			if ( provider.dismissed_errors?.includes( errorType ) ) {
```

Comment:
`errorType` is considered a string here, so we get a TS error:

```
Argument of type 'string' is not assignable to parameter of type '"SuccessTargetError" | "UrlError" | "HttpError" | "UnknownError" | "CrossDomainError" | "LoadTimeoutError" | "RedirectError" | "UrlVerifyError" | "EmptyCSSError" | "XFrameDenyError" | "InvalidURLError" | "ProviderError" | `HttpError-${number}`'.ts(2345)
```

Initially, we had:

```
const errorTypeGroups = Object.entries( errorsByType ) as [
	Critical_CSS_Error_Type,
	CriticalCssErrorDetails[],
][];
```

to avoid type casting `errorType` multiple times 😅.

I guess that still works and we can just cast it here as well.